v2/pkg/protocols/network/request.go

```diff
@@ -274,14 +277,29 @@ func (request *Request) executeRequestWithPayloads(actualAddress, address, input
 		})
 	}
 
-	if request.options.Options.Debug || request.options.Options.DebugResponse {
-		gologger.Debug().Msgf("[%s] Dumped Network response for %s", request.options.TemplateID, actualAddress)
-		gologger.Print().Msgf("%s\nHex: %s", response, responsehighlighter.Highlight(event.OperatorsResult, hex.EncodeToString([]byte(response)), request.options.Options.NoColor))
-	}
+	debug(event, request, response, actualAddress)
 
 	return nil
 }
 
+func debug(event *output.InternalWrappedEvent, request *Request, response string, actualAddress string) {
+	if request.options.Options.Debug || request.options.Options.DebugResponse {
```

Comment:
we could reverse the condition here to remove this indentation level since it makes the code look kinda ugly. This is relevant for other similar debug functions in the PR.

Reply:
Generally speaking, from a readability point of view, negated conditions are harder to read/understand, especially if there are multiple conditions. Normally (even though GO "standards" recommend otherwise), having single return points in methods are easier to follow and understand. If you end up having a lot of indentations, it means that you might need to consider splitting the logic into smaller methods.